File: crates/klyron_bench/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_bench {
BenchBuilder::BenchBuilder() : config_() {}
BenchBuilder& BenchBuilder::enabled(bool v) { config_.enabled = v; return *this; }
BenchBuilder& BenchBuilder::verbose(bool v) { config_.verbose = v; return *this; }
BenchConfig BenchBuilder::build() { return config_; }
}

File: crates/klyron_bench/bindings/rust/builder.rs
use crate::config::BenchConfig;

pub struct BenchBuilder {
    config: BenchConfig,
}

impl BenchBuilder {
    pub fn new() -> Self { Self { config: BenchConfig::default() } }
    pub fn build(self) -> BenchConfig { self.config }
}

impl Default for BenchBuilder {
    fn default() -> Self { Self::new() }
}
